src/plan_partialize.c

```diff
@@ -0,0 +1,235 @@
+{
+	Query	   *parse = root->parse;
+	PartializeWalkerState state = {
+		.found_partialize = false,
+		.looking_for_agg = false,
+	};
+	ListCell   *lc;
+
+	if (CMD_SELECT != parse->commandType)
+		return;
+
+	expression_tree_walker((Node *) parse->targetList, partialize_function_call_walker, &state);
+
+	if (state.found_partialize)
+	{
+		expression_tree_walker((Node *) parse->targetList, ensure_only_partials, NULL);
+
+		foreach(lc, input_rel->pathlist)
+		{
+			Path	   *path = lfirst(lc);
+
+			if (IsA(path, AggPath))
+				((AggPath *) path)->aggsplit = AGGSPLIT_INITIAL_SERIAL;
+		}
+
+		foreach(lc, output_rel->pathlist)
+		{
+			Path	   *path = lfirst(lc);
+
+			if (IsA(path, AggPath))
+				((AggPath *) path)->aggsplit = AGGSPLIT_INITIAL_SERIAL;
+		}
+	}
+}
+
+/* ///////////////////////////////////// */
+
+Datum
+ts_aggregate_transition_type(PG_FUNCTION_ARGS)
+{
+	HeapTuple	aggTuple;
+	Form_pg_aggregate aggform;
+	Oid			trans_type;
+	Oid			agg_fn_oid = PG_GETARG_OID(0);
+
+	/* fetch aggregate entry from pg_aggregate */
+	aggTuple = SearchSysCache1(AGGFNOID, ObjectIdGetDatum(agg_fn_oid));
+	if (!HeapTupleIsValid(aggTuple))
+		elog(ERROR, "invalid aggregate function '%s'", get_func_name(agg_fn_oid));
+
+	aggform = (Form_pg_aggregate) GETSTRUCT(aggTuple);
+	trans_type = aggform->aggtranstype;
+	ReleaseSysCache(aggTuple);
+
+	PG_RETURN_OID(trans_type);
+}
+
+
+Datum
+ts_aggregate_deserialize_fn(PG_FUNCTION_ARGS)
+{
+	HeapTuple	aggTuple;
+	Form_pg_aggregate aggform;
+	Oid			deserialize_fn = InvalidOid;
+	Oid			agg_fn_oid = PG_GETARG_OID(0);
+
+	/* fetch aggregate entry from pg_aggregate */
+	aggTuple = SearchSysCache1(AGGFNOID, ObjectIdGetDatum(agg_fn_oid));
+	if (!HeapTupleIsValid(aggTuple))
+		elog(ERROR, "invalid aggregate function '%s'", get_func_name(agg_fn_oid));
+
+	aggform = (Form_pg_aggregate) GETSTRUCT(aggTuple);
+	if (OidIsValid(aggform->aggdeserialfn))
+		deserialize_fn = aggform->aggdeserialfn;
+	ReleaseSysCache(aggTuple);
+
+	PG_RETURN_OID(deserialize_fn);
+}
+
+Datum
+ts_aggregate_deserialize(PG_FUNCTION_ARGS)
+{
+	bytea	   *serialized = PG_GETARG_BYTEA_P(0);
+	RegProcedure agg = PG_GETARG_OID(1);
+	Oid			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 2);
+	HeapTuple	aggTuple;
+	Form_pg_aggregate aggform;
+	Datum		deserialized;
+
+	aggTuple = SearchSysCache1(AGGFNOID, ObjectIdGetDatum(agg));
+	if (!HeapTupleIsValid(aggTuple))
+		elog(ERROR, "invalid aggregate function '%s'", get_func_name(agg));
+
+	aggform = (Form_pg_aggregate) GETSTRUCT(aggTuple);
+	if (arg_type != aggform->aggtranstype)
+		/* elog(ERROR, "invalid return type, got %s expected %s", ); */
+		elog(ERROR, "invalid return type, got %d expected %d", arg_type, aggform->aggtranstype);
+
+	if (OidIsValid(aggform->aggdeserialfn))
+	{
+		FmgrInfo	deserialize_finfo;
+
+		fmgr_info(aggform->aggdeserialfn, &deserialize_finfo);
+
+		deserialized = FunctionCall1Coll(&deserialize_finfo, InvalidOid, PointerGetDatum(serialized));
+	}
```

Comment:
If we decide we want postgres to handle deserialization for `internal` datatypes, we simply remove the `if` part of this conditional.
(I suspect this _will_ ultimately end up being the strategy, but we need to dig a little deeper into `Finalize Aggregate` to be sure)